tests/test_e2e.py

```diff
@@ -0,0 +1,71 @@
+import random
+import string
+import time
+
+from dewy.chunks.models import RetrieveRequest, RetrieveResponse
+from dewy.documents.models import AddDocumentRequest
+
+SKELETON_OF_THOUGHT_PDF = "https://arxiv.org/pdf/2307.15337.pdf"
+
+
+async def create_collection(client, text_embedding_model: str) -> int:
+    name = "".join(random.choices(string.ascii_lowercase, k=5))
+    create_response = await client.put("/api/collections/", json={"name": name})
+    assert create_response.status_code == 200
+
+    return create_response.json()["id"]
+
+
+async def ingest(client, collection: int, url: str) -> int:
+    add_request = AddDocumentRequest(collection_id=collection, url=url)
+    add_response = await client.put(
+        "/api/documents/", data=add_request.model_dump_json()
+    )
+    assert add_response.status_code == 200
+
+    document_id = add_response.json()["id"]
+
+    # TODO(https://github.com/DewyKB/dewy/issues/34): Move waiting to the server
+    # and eliminate need to poll.
+    status = await client.get(f"/api/documents/{document_id}")
+    while status.json()["ingest_state"] != "ingested":
+        time.sleep(1)
+        status = await client.get(f"/api/documents/{document_id}")
+
+    return document_id
+
+
+async def retrieve(client, collection: int, query: str) -> RetrieveResponse:
+    request = RetrieveRequest(
+        collection_id=collection, query=query, include_image_chunks=False
+    )
+
+    response = await client.post("/api/chunks/retrieve", data=request.model_dump_json())
+    assert response.status_code == 200
+    return RetrieveResponse.model_validate_json(response.content)
+
+
+async def test_e2e_openai_ada002(client):
```

Comment:
Is there a file missing or something? Where does `client` come from?

Reply:
Pytest decides that `conftest.py` is always loaded. `client` is a fixture provided by that.